### PR TITLE
F/remove sklarn

### PR DESCRIPTION
**Pull request by wooohoooo**
+ removed dependency on Sklearn etc, in setup.py and all the places where I could find it
+ removed run() and other pieces of the API that aren't supposed to be used any more
+ made sure the tests cover the use cases
+ added notebook that showcases the use of the package (can be used as secondary testing)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,8 +3,7 @@
 [![PyPI Release](https://github.com/scailable/sclblpy/workflows/PyPI%20Release/badge.svg)](https://pypi.org/project/sclblpy/)
 
 
-Changelog: the package is currently undergoing a major rework. We try to keep the below up to date, but some features might be stubs. 
-Most relevant is the removal of support for Scikit learn in favor of focussing fully on ONNX.
+Changelog: the package is currently being refactored. As such, it is currently a stub, aas Scailable has changed its focus from Scikit Learn to ONNX, Scikit Learn support has been removed.
```
Comment (ayoub-assis) [inline README.md:6]: A minor typo `aas`

File: test/test_main.py [hunk 2]
```diff
@@ -258,16 +187,31 @@ def test_user_utils():
     print("Running simple functional tests of main.py")
     print("===============================")
 
-    test_upload()  # upload sklearn and onnx test
+    test_upload_onnx()  # upload sklearn and onnx test
 
     print("Wait, toolchain needs to finish....")
-    time.sleep(10)
+    time.sleep(25)
+    print(" waited for 10 seconds")
```
Comment (ayoub-assis) [inline test/test_main.py:194]: 10 should be updated to 25

**Comment by ayoub-assis** [comment]
Everything looks fine to me.
Albeit I couldn't run the notebook and test files since I don't have an account with a device.
But maybe that's something that is irrelevant to this PR.
@robinvanemden

**Comment by MKaptein** [comment]
You should definitely setup your own account and create a device so you can test!

File: README.md [hunk 3]
```diff
@@ -17,28 +16,28 @@ Most relevant is the removal of support for Scikit learn in favor of focussing f
...
 The sclblpy package allows users with a valid scailable account (get one at [https://admin.sclbl.net/signup.html](https://admin.sclbl.net/signup.html)) 
 to upload fitted ML / AI models to the Scailable toolchain server. This will result in:
 
 1. The model being tested for errors on the client side.
 2. The model being uploaded to Scailable, tested again, and if all test pass it will be converted to [WebAssembly](https://webassembly.org).
-3. The model being made available to deploy to a pre-regsitered (link on how to do that) device
+3. The model being made available to deploy to a preregistered (link on how to do that) device
 4. (upcoming) The model being testable through test_run(example_input, device_id)
 
 ## Getting started
```
Comment (ayoub-assis) [inline README.md:43]: `upload` should be replaced with `upload_onnx` function

**Review by ayoub-assis — COMMENTED** [review]
The [`run`](https://github.com/scailable/sclblpy/blob/f/remove_sklarn/sclblpy/main.py#L772) function should be removed

File: README.md [hunk 4]
```diff
@@ -178,7 +177,7 @@ the following packages:
 * `sklearn` (legacy, to be removed in future updates)
```
Comment (ayoub-assis) [inline README.md:177]: `sklearn` is no longer a package dependency

File: test/test_main.py [hunk 5]
```diff
@@ -1,91 +1,52 @@
...
-ADMIN_URL = "http://localhost:8008"  # Location of admin for this test
-TOOLCHAIN_URL = "http://localhost:8010"  # Location of toolchain for this test
-TASKMANAGER_URL = "http://localhost:8080"
+ADMIN_URL = "https://usermanager.sclbl.net"  # Location of admin for this test
+TOOLCHAIN_URL = "https://toolchain.sclbl.net"  # Location of toolchain for this test
 
 
-def test_upload():
+def test_upload_onnx():
     """ Test the upload function (sklearn and onnx"""
```
Comment (ayoub-assis) [inline test/test_main.py:17]: Here

File: test/test_main.py [hunk 6]
```diff
@@ -1,91 +1,52 @@
...
     # ONNX
     docs['name'] = "Name of ONNX model"
     docs['documentation'] = "A long .md thing...."
-    check = upload("../test/files/model.onnx", "", docs, model_type="onnx")
+    check = upload_onnx("../test/files/model.onnx", "", docs)
     assert check is True, "ONNX upload test failed."
 
 
 def test_update():
     """ Test the update function (sklearn & onnx"""
```
Comment (ayoub-assis) [inline test/test_main.py:29]: Here too

**Comment by ayoub-assis** [comment]
- The package documentation should be regenerated: [main.m.html](https://github.com/scailable/sclblpy/blob/f/remove_sklarn/docs/main.m.html)
- This file is irrelevant now [supported_models.json](https://github.com/scailable/sclblpy/blob/f/remove_sklarn/sclblpy/supported_models.json)
- docstring of [`test_run_inputs'](https://github.com/scailable/sclblpy/blob/81d6ca7167b33ae835a0260d5c85c6fa0b39397e/test/test_run.py#L12)

File: test/test_main.py [hunk 7]
```diff
@@ -258,16 +187,31 @@ def test_user_utils():
     print("Running simple functional tests of main.py")
     print("===============================")
 
-    test_upload()  # upload sklearn and onnx test
+    test_upload_onnx()  # upload sklearn and onnx test
```
Comment (ayoub-assis) [inline test/test_main.py:190]: `upload sklearn and onnx test` ---> `upload onnx test`

File: test/test_main.py [hunk 8]
```diff
@@ -258,16 +187,31 @@ def test_user_utils():
...
 
     print("Wait, toolchain needs to finish....")
-    time.sleep(10)
+    time.sleep(25)
+    print(" waited for 10 seconds")
 
+    print('testing update \n\n')
     test_update()  # update sklearn and onnx test
```
Comment (ayoub-assis) [inline test/test_main.py:197]: `update sklearn and onnx test` --> `update onnx test`

**Comment by ayoub-assis** [comment]
## Manual Testing
- test_utils: passed
- test_bundle: passed
- test_main: passed

- Notebook works fine

**Comment by ayoub-assis** [comment]
- test_jwt: Passed after specifying the right `usermanager` server was correctly specified

**Comment by robinvanemden** [comment]
Works for me locally. All issues that were found were also resolved by @ayoub-assis .
Merging.

**Pull request merged**